--- a/config/BuildSystem/config/packages/petsc4py.py	2021-03-21 01:35:48.429692000 -0000
+++ b/config/BuildSystem/config/packages/petsc4py.py	2021-03-21 02:14:56.154551000 -0000
@@ -14,7 +14,7 @@
 
   def setupHelp(self,help):
     import nargs
-    help.addArgument('PETSc', '-with-petsc4py=<bool>', nargs.Arg(None, 0, 'Build PETSc Python bindings (petsc4py)'))
+    help.addArgument('PETSc', '-with-petsc4py=<bool>', nargs.ArgBool(None, False, 'Build PETSc Python bindings (petsc4py)'))
     help.addArgument('PETSc', '-with-petsc4py-test-np=<np>',nargs.ArgInt(None, None, min=1, help='Number of processes to use for petsc4py tests'))
     return
 
