--- a/setup.py	2021-03-20 03:01:50.218343000 -0000
+++ b/setup.py	2021-03-20 03:02:57.529333000 -0000
@@ -82,11 +82,17 @@
       author="Imperial College London and others",
       author_email="firedrake@imperial.ac.uk",
       url="http://firedrakeproject.org",
-      packages=["firedrake", "firedrake.mg", "firedrake.slope_limiter",
-                "firedrake.matrix_free", "firedrake.preconditioners",
+      packages=["firedrake",
+                "firedrake.adjoint",
                 "firedrake.cython",
+                "firedrake.mg",
+                "firedrake.slope_limiter",
+                "firedrake.matrix_free",
+                "firedrake.preconditioners",
                 "firedrake.slate", "firedrake.slate.slac", "firedrake.slate.static_condensation",
-                "firedrake_configuration", "firedrake_citations"],
+                "firedrake_configuration",
+                "firedrake_citations"
+                ],
       package_data={"firedrake": ["evaluate.h",
                                   "locate.c",
                                   "icons/*.png"]},
