--- a/setup.py	2021-03-22 17:16:42.622424515 -0000
+++ b/setup.py	2021-03-22 17:16:50.822125317 -0000
@@ -1,4 +1,5 @@
 from distutils.core import setup
+from setuptools import find_packages
 from glob import glob
 from os import environ as env, path
 from Cython.Distutils import build_ext
@@ -82,11 +83,7 @@
       author="Imperial College London and others",
       author_email="firedrake@imperial.ac.uk",
       url="http://firedrakeproject.org",
-      packages=["firedrake", "firedrake.mg", "firedrake.slope_limiter",
-                "firedrake.matrix_free", "firedrake.preconditioners",
-                "firedrake.cython",
-                "firedrake.slate", "firedrake.slate.slac", "firedrake.slate.static_condensation",
-                "firedrake_configuration", "firedrake_citations"],
+      packages=find_packages(),
       package_data={"firedrake": ["evaluate.h",
                                   "locate.c",
                                   "icons/*.png"]},
