--- a/config/prte_check_alps.m4	2021-04-11 02:33:59.000000000 -0000
+++ b/config/prte_check_alps.m4	2021-04-11 02:38:31.000000000 -0000
@@ -149,6 +149,18 @@
                                                        AC_MSG_WARN([on the configure line using --with-alps option.])
                                                        AC_MSG_ERROR([Aborting])],[])]
                                                        )
+                               PKG_CHECK_MODULES_STATIC([CRAY_UGNI], [cray-ugni],
+                                               [prte_check_cray_alps_happy="yes"
+                                                AC_DEFINE_UNQUOTED([CRAY_UGNI],[1],
+                                                                   [defined to 1 if cray uGNI available, 0 otherwise])
+                                               ],
+                                               [prte_check_cray_alps_happy="no"]
+                                               [AS_IF([test "$with_alps" = "yes"],
+                                                      [AC_MSG_WARN([ALPS support requested but pkg-config failed.])
+                                                       AC_MSG_WARN([Need to explicitly indicate ALPS directory])
+                                                       AC_MSG_WARN([on the configure line using --with-alps option.])
+                                                       AC_MSG_ERROR([Aborting])],[])]
+                                                       )
                             ],
                             [AC_MSG_WARN([See ./configure --help for how to control Open MPI])
                              AC_MSG_WARN([configuration for ALPS on CLE 5 and higher systems])
@@ -177,9 +189,9 @@
     AS_IF([test "$prte_check_cray_alps_happy" = "yes"],
           [$1_LDFLAGS="[$]$1_LDFLAGS $CRAY_ALPSLLI_LIBS $CRAY_ALPSUTIL_LIBS"
            $1_CPPFLAGS="[$]$1_CPPFLAGS $CRAY_ALPSLLI_CFLAGS $CRAY_ALPSUTIL_CFLAGS $CRAY_ALPS_CFLAGS $CRAY_WLM_DETECT_CFLAGS"
-           $1_LIBS="[$]$1_LIBS $CRAY_ALPSLLI_LIBS $CRAY_ALPSUTIL_LIBS $CRAY_WLM_DETECT_LIBS"
-           $1_WRAPPER_EXTRA_LDFLAGS="$CRAY_ALPSLLI_LIBS $CRAY_ALPSUTIL_LIBS $CRAY_WLM_DETECT_LIBS"
-           $1_WRAPPER_EXTRA_LIBS="$CRAY_ALPSLLI_LIBS $CRAY_ALPSUTIL_LIBS $CRAY_WLM_DETECT_LIBS"
+           $1_LIBS="[$]$1_LIBS $CRAY_ALPSLLI_LIBS $CRAY_ALPSUTIL_LIBS $CRAY_WLM_DETECT_LIBS $CRAY_UGNI_LIBS"
+           $1_WRAPPER_EXTRA_LDFLAGS="$CRAY_ALPSLLI_LIBS $CRAY_ALPSUTIL_LIBS $CRAY_WLM_DETECT_LIBS $CRAY_UGNI_LIBS"
+           $1_WRAPPER_EXTRA_LIBS="$CRAY_ALPSLLI_LIBS $CRAY_ALPSUTIL_LIBS $CRAY_WLM_DETECT_LIBS $CRAY_UGNI_LIBS"
 	   $2],
 	  [$3])
 ])
