--- a/3rd-party/prrte/src/mca/plm/lsf/plm_lsf_module.c	2021-03-02 00:47:25.066312000 -0000
+++ b/3rd-party/prrte/src/mca/plm/lsf/plm_lsf_module.c	2021-03-02 00:47:31.667233000 -0000
@@ -388,7 +388,7 @@
 
     /* check for failed launch - if so, force terminate */
     if (failed_launch) {
-        PRTE_ACTIVATE_JOB_STATE(state->jata, PRTE_JOB_STATE_FAILED_TO_START);
+        PRTE_ACTIVATE_JOB_STATE(state->jdata, PRTE_JOB_STATE_FAILED_TO_START);
     }
 
     /* cleanup the caddy */
