--- a/./config/opal_mca.m4	2021-04-11 06:01:48.000000000 -0000
+++ b/./config/opal_mca.m4	2021-04-11 06:03:42.000000000 -0000
@@ -68,7 +68,8 @@
                         type-component pairs that will be built as
                         run-time loadable components (as opposed to
                         statically linked in), if supported on this
-                        platform.])])
+                        platform.])],
+                        [], [enable_mca_dso=btl-ugni,rcache-udreg])
     AC_ARG_ENABLE([mca-static],
         [AS_HELP_STRING([--enable-mca-static=LIST],
                        [Comma-separated list of types and/or
