--- a/3rd-party/prrte/src/tools/prte/prte.c	2021-03-03 22:55:52.786727000 -0000
+++ b/3rd-party/prrte/src/tools/prte/prte.c	2021-03-03 22:57:58.997749000 -0000
@@ -348,8 +348,8 @@
     /* Mapping options */
     { '\0', "map-by", 1, PRTE_CMD_LINE_TYPE_STRING,
       "Mapping Policy for job [slot | hwthread | core (default:np<=2) | l1cache | "
-      "l2cache | l3cache | package (default:np>2) | node | seq | dist | ppr |,"
-      "rankfile:<rankfile_path>]"
+      "l2cache | l3cache | package (default:np>2) | node | seq | dist | ppr | "
+      "rankfile:FILE=<rankfile_path>]"
       " with supported colon-delimited modifiers: PE=y (for multiple cpus/proc), "
       "SPAN, OVERSUBSCRIBE, NOOVERSUBSCRIBE, NOLOCAL, HWTCPUS, CORECPUS, "
       "DEVICE(for dist policy), INHERIT, NOINHERIT, PE-LIST=a,b (comma-delimited "
