--- a/3rd-party/prrte/src/util/nidmap.c	2021-03-03 17:39:11.613332000 -0000
+++ b/3rd-party/prrte/src/util/nidmap.c	2021-03-03 17:39:24.882222000 -0000
@@ -783,14 +783,14 @@
         /* now get the array of assigned topologies */
         /* unpack the compression flag */
         cnt = 1;
-        rc = PMIx_Data_unpack(NULL, &bucket, &compressed, &cnt, PMIX_BOOL);
+        rc = PMIx_Data_unpack(NULL, buf, &compressed, &cnt, PMIX_BOOL);
         if (PMIX_SUCCESS != rc) {
             PMIX_ERROR_LOG(rc);
             goto cleanup;
         }
         /* unpack the topologies object */
         cnt = 1;
-        rc = PMIx_Data_unpack(NULL, &bucket, &pbo, &cnt, PMIX_BYTE_OBJECT);
+        rc = PMIx_Data_unpack(NULL, buf, &pbo, &cnt, PMIX_BYTE_OBJECT);
         if (PMIX_SUCCESS != rc) {
             PMIX_ERROR_LOG(rc);
             goto cleanup;
